pages/docs/integrations/quarkus-langchain4j.mdx

```diff
@@ -0,0 +1,99 @@
+
+```xml
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-opentelemetry</artifactId>
+</dependency>
+```
+
+**Configure OpenTelemetry exporter** (`application.properties`):
+
+```properties
+quarkus.otel.exporter.otlp.traces.protocol=http/protobuf
+```
+
+With these configurations and dependencies in place, your Quarkus application is ready to produce OpenTelemetry traces. Quarkus LangChain4j internal calls (e.g. when you invoke a chat model) will be recorded as spans.
+
+Each span will carry attributes like `gen_ai.operation.name`, `gen_ai.system` (the provider, e.g. “openai”), model names, token usage, etc., and – since we enabled them – events for the prompt and response content
+
+## Step 2: Configure Langfuse
+
+Now that your Quarkus application is emitting OpenTelemetry trace data, the next step is to direct that data to Langfuse.
+
+Langfuse will act as the “backend” for OpenTelemetry in this setup – essentially replacing a typical Jaeger/Zipkin/OTel-Collector with Langfuse’s trace ingestion API.
+
+
+**Langfuse Setup**
+   - Sign up for [Langfuse Cloud](https://cloud.langfuse.com/) or [self-hosted Langfuse](https://langfuse.com/self-hosting).
+   - Set the OTLP endpoint (e.g. `https://cloud.langfuse.com/api/public/otel`) and API keys.
+
+Configure these via environment variables:
+
+```bash
+QUARKUS_OTEL_EXPORTER_OTLP_ENDPOINT: set this to the Langfuse OTLP URL (e.g. https://cloud.langfuse.com/api/public/otel).
+QUARKUS_OTEL_EXPORTER_OTLP_HEADERS: set this to Authorization=Basic <base64 public:secret>.
+```
+
+<Callout type="info">
+You can find more on authentication via Basic Auth [here](/docs/opentelemetry/get-started).
+</Callout>
+
+## Step 3: Run a Test AI Operation
+
+Start your Quarkus application. Trigger an AI operation that Quarkus LangChain4j handles – for example, call a service or controller that uses a `ChatModel` to generate a completion.
+
+```java
+@RegisterAiService(tools = EmailService.class)
+public interface MyAiService {
+
+    /**
+     * Ask the LLM to create a poem about the given topic.
+     *
+     * @param topic the topic of the poem
+     * @param lines the number of line of the poem
+     * @return the poem
+     */
+    @SystemMessage("You are a professional poet")
+    @UserMessage("""
+            Write a single poem about {topic}. The poem should be {lines} lines long and your response should only include the poem itself, nothing else.
+            Then send this poem by email. Your response should include the poem.
+            """)
+    String writeAPoem(String topic, int lines);
+
+}
+
+@Singleton
+public class Startup {
+
+    public void writeAPoem(@Observes StartupEvent event, MyAiService service) {
+        System.out.println(service.writeAPoem("LangFuse", 4));
+    }
+}
+```
+
+## Troubleshooting
+
+**No Traces:**
+  - Check the logs of the application for potential clues
```

Comment:
You can reference our regular troubleshooting page here as well: https://langfuse.com/self-hosting/troubleshooting#missing-events-after-post-apipublicingestion.